python/cuml/preprocessing/model_selection.py

```diff
@@ -93,8 +93,10 @@ def train_test_split(
     if seed is not None:
         np.random.seed(seed)
 
+    # Replace Numpy/cuDF here when issue mentioned above is solved!
     if shuffle:
-        idxs = _shuffle_idx(X)
+        idxs = cudf.utils.cudautils.arange(len(X)).copy_to_host()
```

Comment:
Maybe just numpy.arange since you're going to copy to host anyways? Seems like we should either try for all on-device or just accept numpy.

Reply:
Fair point. Hopefully this will be cupy.arange instead pretty soon :)